sync-core/src/test/java/com/cloudant/sync/query/QueryTextSearchTest.java

```diff
@@ -0,0 +1,233 @@
+    public void canMakeORCompoundQueryWithATextSearch() {
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("name"), "basic"), is("basic"));
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("comment"), "basic_text", "text"),
+                   is("basic_text"));
+
+        // query - { "$or" : [ { "name" : "mike" }, { "$text" : { "$search" : "best friend" } } ] }
+        Map<String, Object> search = new HashMap<String, Object>();
+        search.put("$search", "best friend");
+        Map<String, Object> text = new HashMap<String, Object>();
+        text.put("$text", search);
+        Map<String, Object> name = new HashMap<String, Object>();
+        name.put("name", "mike");
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("$or", Arrays.<Object>asList(name, text));
+
+        QueryResult queryResult = im.find(query);
+        assertThat(queryResult.documentIds(), containsInAnyOrder("mike12",
+                                                                 "mike34",
+                                                                 "mike72",
+                                                                 "fred12"));
+    }
+
+    @Test
+    public void nullForTextSearchQueryWithoutATextIndex() {
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("name"), "basic"), is("basic"));
+
+        // query - { "name" : "mike", "$text" : { "$search" : "best friend" } }
+        Map<String, Object> search = new HashMap<String, Object>();
+        search.put("$search", "best friend");
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        query.put("$text", search);
+        assertThat(im.find(query), is(nullValue()));
+    }
+
+    @Test
+    public void nullForTextSearchQueryWhenAJsonIndexIsMissing() {
+        // All fields in a TEXT index only apply to the text search portion of any query.
+        // So even though "name" exists in the text index, the clause that { "name" : "mike" }
+        // expects a JSON index that contains the "name" field.  Since, this query includes a
+        // text search clause then all clauses of the query must be satisfied by existing indexes.
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("name", "comment"), "basic_text", "text"),
+                   is("basic_text"));
+
+        // query - { "$or" : [ { "name" : "mike" }, { "$text" : { "$search" : "best friend" } } ] }
+        Map<String, Object> search = new HashMap<String, Object>();
+        search.put("$search", "best friend");
+        Map<String, Object> text = new HashMap<String, Object>();
+        text.put("$text", search);
+        Map<String, Object> name = new HashMap<String, Object>();
+        name.put("name", "mike");
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("$or", Arrays.<Object>asList(name, text));
+
+        assertThat(im.find(query), is(nullValue()));
+    }
```

Comment:
In addition, try the porter tokeniser out.

- A search for `live` should fail with the default stemmer (because it's not a prefix search) but should succeed with porter (as `live` is a stem of `lives`).
- Ditto for `retire`.